metagraph/src/graph/alignment/aligner_aggregator.hpp

```diff
@@ -22,9 +26,23 @@ class PriorityDeque : public boost::container::priority_deque<Type, Sequence, Co
 template <class AlignmentCompare>
 class AlignmentAggregator {
   public:
+    struct SharedPtrCmp {
+        bool operator()(const std::shared_ptr<Alignment> &a,
+                        const std::shared_ptr<Alignment> &b) const {
+            return base_cmp_(*a, *b);
+        }
+
+        AlignmentCompare base_cmp_;
+    };
+
     typedef Alignment::node_index node_index;
     typedef Alignment::score_t score_t;
-    typedef PriorityDeque<Alignment, std::vector<Alignment>, AlignmentCompare> PathQueue;
+    typedef annot::binmat::BinaryMatrix::Column Column;
+    typedef PriorityDeque<std::shared_ptr<Alignment>,
+                          std::vector<std::shared_ptr<Alignment>>,
+                          SharedPtrCmp> PathQueue;
```

Comment:
Depends on how much time `std::move` takes for the objects, but usually avoiding pointers and storing the objects with implemented `std::move` is significantly faster. I would suggest benchmarking this.

Reply:
The idea behind this is to prevent keeping several copies of the same alignment if there's a path which is shared among several labels.
`AlignmentAggregator` operations haven't been a bottleneck, so I didn't think that this needed any optimization.

Reply:
Isn't a priority queue critical to the performance of an alignment algorithm?

Reply:
Hm, ok. Why don't you just push all the alignments into a vector and sort them? Why do you need a queue here at all?